.github/workflows/workflow.yml

```diff
@@ -9,48 +9,68 @@ on:
+          sudo apt-get autoremove
+          sudo apt-get autoclean
+          pip install pipenv
+          pip install tox
+          # install IPFS
+          sudo apt-get install -y wget
+          wget -O ./go-ipfs.tar.gz https://dist.ipfs.io/go-ipfs/v0.4.23/go-ipfs_v0.4.23_linux-amd64.tar.gz
+          tar xvfz go-ipfs.tar.gz
+          sudo mv go-ipfs/ipfs /usr/local/bin/ipfs
+          ipfs init
+      - name: Security Check
+        run: tox -e bandit
+      - name: Safety Check
+        run: tox -e safety
+      - name: License Check
+        run: tox -e liccheck
+      - name: Copyright Check
+        run: tox -e copyright_check
+      - name: AEA Package Hashes Check
+        run: tox -e hash_check -- --timeout 20.0
+      - name: Code style check
+        run: |
+          tox -e black-check
+          tox -e flake8
+      - name: Static type check
+        run: tox -e mypy
+      - name: Check package versions in documentation
+        run: tox -e package_version_checks
+      - name: Generate Documentation
+        run: tox -e docs
+
+  integration_checks:
+    runs-on: ubuntu-latest
 
-     steps:
-       - uses: actions/checkout@master
-       - uses: actions/setup-python@master
-         with:
-           python-version: 3.6
-       - name: Install dependencies (ubuntu-latest)
-         run: |
-           sudo apt-get update --fix-missing
-           sudo apt-get autoremove
-           sudo apt-get autoclean
-           pip install pipenv
-           pip install tox
-           # install IPFS
-           sudo apt-get install -y wget
-           wget -O ./go-ipfs.tar.gz https://dist.ipfs.io/go-ipfs/v0.4.23/go-ipfs_v0.4.23_linux-amd64.tar.gz
-           tar xvfz go-ipfs.tar.gz
-           sudo mv go-ipfs/ipfs /usr/local/bin/ipfs
-           ipfs init
-       - name: Security Check
-         run: tox -e bandit
-       - name: Safety Check
-         run: tox -e safety
-       - name: License Check
-         run: tox -e liccheck
-       - name: Copyright Check
-         run: tox -e copyright_check
-       - name: AEA Package Hashes Check
-         run: tox -e hash_check -- --timeout 20.0
-       - name: Code style check
-         run: |
-           tox -e black-check
-           tox -e flake8
-       - name: Static type check
-         run: tox -e mypy
-       - name: Check package versions in documentation
-         run: tox -e package_version_checks
-       - name: Generate Documentation
-         run: tox -e docs
+    timeout-minutes: 30
+
+    steps:
+      - uses: actions/checkout@master
+      - uses: actions/setup-python@master
+        with:
+          python-version: 3.7
+      - name: Install dependencies (ubuntu-latest)
+        run: |
+          sudo apt-get update --fix-missing
+          sudo apt-get autoremove
+          sudo apt-get autoclean
+          pip install pipenv
+          pip install tox
+      - name: Integration tests
+        run: tox -e py3.7 -- -m integration
```

Comment:
@solarw we now run integration tests separately here. To mark a test as integration test we do `@pytest.mark.integration`. We still need to make a choice which tests qualify for this versus the platform checks

Reply:
For now, we only run them on ubuntu-latest 3.7. They take a long time to run and require actual resources (test tokens), so we should not overdo it.